src/libraries/Microsoft.PowerFx.Interpreter/Environment/ComposedReadOnlySymbolValues.cs

```diff
@@ -62,16 +57,26 @@ internal class ComposedReadOnlySymbolValues : ReadOnlySymbolValues
                             throw new InvalidOperationException($"ComposedSymbolValues should have been flattened ");
                         }
                         
-                        Add(map, kv.Value);
+                        Add(existingMap, kv.Value);
                     }
                 }
                 else
                 {
-                    Add(map, symValues);
+                    Add(existingMap, symValues);
                 }                
             }
-            
-            CreateValues(canCreateNew, map, symbolTable);
+
+            // CreateValues will remove from existingMap, so anything left is extra.
+            var map = new Dictionary<ReadOnlySymbolTable, ReadOnlySymbolValues>();
+            CreateValues(canCreateNew, existingMap, map, symbolTable);
+
+            if (existingMap.Count > 0)
```

Comment:
This is a very strong invariant to enforce - it ensures that we don't pass in "extra" symbol values that aren't matched with a table.  With this refactoring, there is no longer any reason for a caller to do that. 
(where as before, a caller had to do this to include a ServiceProvider).